ui/packages/components/src/Timeline/TimelineNode/TimelineNode.tsx

```diff
@@ -107,20 +127,52 @@ function Content({
     <>
       <div className="pb-5">
         <MetadataGrid
-          metadataItems={[
-            {
-              label: 'Started At',
-              value: node.scheduledAt ? node.scheduledAt.toLocaleString() : '-',
-            },
-            {
-              label: 'Ended At',
-              value: node.endedAt ? node.endedAt.toLocaleString() : '-',
-            },
-            {
-              label: 'Duration',
-              value: durationMS ? formatMilliseconds(durationMS) : '-',
-            },
-          ]}
+          metadataItems={
+            [
+              {
+                label: 'Started At',
+                value: node.scheduledAt ? node.scheduledAt.toLocaleString() : '-',
+                title: node?.scheduledAt?.toLocaleString(),
+              },
+              {
+                label: endedAtLabel,
+                value: node.endedAt ? node.endedAt.toLocaleString() : '-',
+                title: node?.endedAt?.toLocaleString(),
+              },
+              {
+                label: 'Duration',
+                value: durationMS ? formatMilliseconds(durationMS) : '-',
+                tooltip: `Time between ${type ? type : 'step'} started and ${
+                  type ? type : 'step'
+                } ${tootltipLabel}`,
+              },
+              ...(node.sleepConfig?.until
+                ? [
+                    {
+                      label: 'Sleep Until',
+                      value: node.sleepConfig?.until?.toLocaleString(),
+                    },
+                  ]
+                : []),
+              ...(node.waitForEventConfig
+                ? [
+                    {
+                      label: 'Event Name',
+                      value: (
+                        <>
+                          <IconEvent className="inline-block" /> {node.waitForEventConfig.eventName}
+                        </>
+                      ),
+                    },
+                    {
+                      label: 'Match Expression',
+                      value: node.waitForEventConfig.expression ?? 'N/A',
+                      type: 'code',
+                    },
+                  ]
+                : []),
+            ] as MetadataItemProps[]
```

Comment:
Is there a better way to do this @goodoldneon ? I thought I wouldn't even have to do anything, but TypeScript's inference was not enough here.

Reply:
It's because TypeScript is interpreting `type` as a string on line 170 but elsewhere we expect `type` to be a literal union. Adding `as const` on line 171 should obviate the need for type casting